### PR TITLE
A4A > Partner Directory: Add new industry values to PD

### PR DESCRIPTION
**Pull request by yashwin**
Related to https://github.com/Automattic/automattic-for-agencies-dev/issues/1116

## Proposed Changes

This PR updates the industry values for the Partner Directory.

## Why are these changes being made?

* To update the industry values for the Partner Directory.

## Testing Instructions

* Update the your Partner Directory using the production to have all all the available industries. 
* Open the A4A live link
* Verify that the old values are mapped to the new ones mentioned below. Also, the new values should be able to be selected. Go ahead and save the changes and verify it works as expected without breaking anything.

Newly added values: 

Technology & IT Services
E-commerce & Retail
Finance & Insurance
Healthcare & Medical
Education & E-Learning
Real Estate & Property
Travel & Hospitality
Nonprofits & NGOs
Legal & Professional Services
Entertainment & Media
Construction & Engineering
Automotive & Transportation
Government & Public Services
Marketing & Advertising
Food & Beverage
Manufacturing & Industry
Energy & Utilities
Sports & Recreation
Agriculture & Farming
Arts & Culture
Environmental & Sustainability
Fashion & Beauty
Logistics & Supply Chain
Events & Conferences

Mapping:

Agricultural services → Agriculture & Farming
Clothing shops → Fashion & Beauty
Government services → Government & Public Services
Miscellaneous shops → E-commerce & Retail
Retail outlet services → E-commerce & Retail
Transportation services → Automotive & Transportation
Utility services → Energy & Utilities
Business services → Legal & Professional Services
Contracted services → Legal & Professional Services
Professional services and membership organisations → Legal & Professional Services

Screenshot 2024-10-01 at 2 55 07 PM

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/a8c-for-agencies/sections/partner-directory/components/hooks/use-form-selectors.ts [hunk 1]
```diff
@@ -58,11 +70,25 @@ export function useFormSelectors() {
...
+		utility_services: 'energy_and_utilities',
+	};
+
 	return {
 		availableServices,
 		availableLanguages,
 		availableProducts,
 		availableDirectories,
-		availableIndustries,
+		availableIndustries: Object.fromEntries( Object.entries( availableIndustries ).sort() ),
```
Comment (vitozev) [inline client/a8c-for-agencies/sections/partner-directory/components/hooks/use-form-selectors.ts]: Is there a specific reason why we are performing the sort operation here? Can't we simply define the items in the order we'd like?

File: client/a8c-for-agencies/sections/partner-directory/components/industries-selector.tsx [hunk 2]
```diff
@@ -31,12 +31,18 @@ const IndustriesSelector = ( { setIndustries, industries }: Props ) => {
...
-
+	const selectedIndustriesByLabel = Array.from(
+		new Set(
+			industries.flatMap( ( slug ) => {
+				const key = slug as string;
+				const value =
+					key in oldIndustries
+						? availableIndustries[ oldIndustries[ key ] ]
+						: availableIndustries[ key ];
+				return value ? [ value ] : [];
```
Comment (vitozev) [inline client/a8c-for-agencies/sections/partner-directory/components/industries-selector.tsx]: Can this be simplified to this line?

```
return availableIndustries[oldIndustries[key] || key];
```

File: client/a8c-for-agencies/sections/partner-directory/components/hooks/use-form-selectors.ts (hunk 1 above)
Comment (yashwin) [reply client/a8c-for-agencies/sections/partner-directory/components/hooks/use-form-selectors.ts]: I thought it's better to sort it here as there is a chance that we might miss updating it alphabetically. I have now made the changes and added a comment: https://github.com/Automattic/wp-calypso/pull/95053/commits/2de9f73edbf90443a0e7c099baf86969b5c6b480

File: client/a8c-for-agencies/sections/partner-directory/components/industries-selector.tsx (hunk 2 above)
Comment (yashwin) [reply client/a8c-for-agencies/sections/partner-directory/components/industries-selector.tsx]: I have update it now: https://github.com/Automattic/wp-calypso/pull/95053/commits/2de9f73edbf90443a0e7c099baf86969b5c6b480

**Comment by yashwin** [comment]
Thanks for the review, @vitozev.

I have addressed your comments. Could you please take another look?

**Review by vitozev — APPROVED** [review]
LGTM!

**Comment by cleacos** [comment]
The plugin was already updated in all of the Partner Directories:

- WooCommerce.com
- Pressable.com
- WordPress.com

Now, we can go ahead and merge this PR.

**Review by cleacos — CHANGES_REQUESTED** [review]
I tested this PR using the life link and if I save my profile clicking on the CTA button it will reload the full page. In production works well.

I cannot identify anything weird in the code.

**Comment by cleacos** [comment]
Taking a random PR in GH, it works well. Maybe this needs to be rebased.

**Comment by cleacos** [comment]
Rebased!

**Comment by cleacos** [comment]
After rebasing it, it works well without that page refresh.

**Comment by cleacos** [comment]
@vitozev Updating the Expertise will turn on the agency's visibility from not visible to visible.

Issue to fix it: https://github.com/Automattic/automattic-for-agencies-dev/issues/1257

**Review by cleacos — APPROVED** [review]
LGTM!

**Pull request merged**